### PR TITLE
Add example as a notebook: example_na_controlling_the_pml

### PR DESCRIPTION
**Pull request by faridyagubbayli**
Adds our first Colab-runnable example :tada: 

Migrates an example (`example_na_controlling_the_pml`) from Matlab to Python as a Jupyter notebook. Changes also include a demonstration of equality between Matlab and Python outputs. A modified Matlab script is available for collecting Matlab outputs.

Note: "Run in Colab" button will be functional upon merging this PR.

### CODE REVIEW

File: examples/example_na_controlling_the_pml/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+</a>
+
+# Running the example
+
+Please have a look at the `example_na_controlling_the_pml.ipynb` notebook file to see the example in action.
+
+## Preparing the Matlab reference files
+
+We can verify the correctness of Python implementation by saving intermediate/final variable states from the Matlab script
+and comparing them against Pyton version. Saving the variable states can be done manually by running the Matlab script,
```
Comment (yfarid-lilt) [inline examples/example_na_controlling_the_pml/README.md:14]: Fix the typo here, thanks @waltsims for catching this!

**Review by waltsims — CHANGES_REQUESTED** [review]
Hey Farid!

Amazing PR. The output plot of the notebook looks like it needs to be fixed.

**Comment by waltsims** [comment]
[This](https://github.com/waltsims/k-wave-python/blob/991c891aa8493e66ec4bc12efb098eb6f018b9d8/examples/at_array_as_sensor/at_array_as_sensor.py#L63) might be the method we need to fix this example. I am leaving this comment for myself later.

**Comment by waltsims** [comment]
> [This](https://github.com/waltsims/k-wave-python/blob/991c891aa8493e66ec4bc12efb098eb6f018b9d8/examples/at_array_as_sensor/at_array_as_sensor.py#L63) might be the method we need to fix this example. I am leaving this comment for myself later.

This indeed worked.

**Comment by waltsims** [comment]
@faridyagubbayli please check again that get_color_map works as intended. perhaps we could make a test so we can test it in all python versions? I was having issues with python 3.8.

File: tests/matlab_test_data_collectors/python_testers/getColorMap_test.py
Comment (waltsims) [inline tests/matlab_test_data_collectors/python_testers/getColorMap_test.py:1]: Please make sure the example follows the passing proper usage and we're good!

**Review by waltsims — APPROVED** [review]

**Comment by waltsims** [comment]
> [This](https://github.com/waltsims/k-wave-python/blob/991c891aa8493e66ec4bc12efb098eb6f018b9d8/examples/at_array_as_sensor/at_array_as_sensor.py#L63) might be the method we need to fix this example. I am leaving this comment for myself later.

This should be updated in the logic kWaveSimulation. There is an option there but it seems the logic doesn't work atm. Same in kwave?

**Comment by faridyagubbayli** [comment]
@waltsims going to merge this soon, please let me know if I should not. I agree that we can fix the method you linked in the next point release.

**Comment by waltsims** [comment]
Im happy to change the color map in the notebook and add 'update colormap' as an enhancement for the next point release. All examples will need it!

**Comment by waltsims** [comment]
I made the change. Just test the notebook please.

**Comment by faridyagubbayli** [comment]
@waltsims thanks for making the change! I tested the notebook. Although colors are different, everything seems fine. Going to merge.

**Pull request merged**